c10/core/Scalar.h

```diff
@@ -158,14 +198,68 @@ class C10_API Scalar {
     }
   }
 
+  Scalar(Scalar&& rhs) noexcept : tag(rhs.tag) {
+    moveFrom(std::move(rhs));
```

Comment:
okay, we don't need `destroy` call here since we are constructing a brand new `Scalar` here